internal/cmd/commands/daemon/search_handler.go

```diff
@@ -99,50 +112,41 @@ func newSearchTargetsHandlerFunc(ctx context.Context, repo *cache.Repository) (h
 	}, nil
 }
 
-func searchTargets(ctx context.Context, repo *cache.Repository, authTokenId, query string, filter *handlers.Filter) (*SearchResult, error) {
-	var found []*targets.Target
-	var err error
-	switch query {
-	case "":
-		found, err = repo.ListTargets(ctx, authTokenId)
-	default:
-		found, err = repo.QueryTargets(ctx, authTokenId, query)
-	}
-	if err != nil {
-		return nil, err
-	}
+type searcher interface {
+	search(ctx context.Context, authTokenId, query string, filter *handlers.Filter) (*SearchResult, error)
+}
 
-	finalTars := make([]*targets.Target, 0, len(found))
-	for _, item := range found {
-		if filter.Match(item) {
-			finalTars = append(finalTars, item)
-		}
-	}
-	return &SearchResult{
-		Targets: finalTars,
-	}, nil
+// searchFns is a struct that collects all the functions needed to perform a search
+// on a specific resource type.
+type searchFns[T any] struct {
+	// list takes a context and an auth token and returns all resources for the
+	// user of that auth token.
+	list func(context.Context, string) ([]T, error)
+	// query takes a context, an auth token, and a query string and returns all
+	// resources for that auth token that matches the provided query parameter
+	query        func(context.Context, string, string) ([]T, error)
+	searchResult func([]T) *SearchResult
 }
 
-func searchSessions(ctx context.Context, repo *cache.Repository, authTokenId, query string, filter *handlers.Filter) (*SearchResult, error) {
-	var found []*sessions.Session
+func (l *searchFns[T]) search(ctx context.Context, authTokenId, query string, filter *handlers.Filter) (*SearchResult, error) {
```

Comment:
perhaps a comment that this implements the `resourceSearcher` interface and what
it does?